### PR TITLE
Allow the date- and time-format strings to contain escaped characters

### LINKED ISSUE #632: Allow backslashes in date format
### Describe the bug

The default date for WordPress in es_ES is "j \d\e F \d\e Y", which should render as "4 de marzo de 2024"

In GatherPress it shows as "4 DDzz marzo DDzz 2024" because it escapes the backslashes in the date format.

### Steps to Reproduce

1. Go to Events > Settings and set the date format as "j \d\e F \d\e Y"
2. Create an event

### Screenshots, screen recording, code snippet

image
image

### Environment information

_No response_

### WordPress information

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by carstingaxion**
The current conversion from php dateformat into javascript moment format is not able to handle escaped characters.

### Description of the Change

My requested change makes the format conversion aware of escaped characters and tells it, to just go on, in such case.

Closes #632

### How to test the Change

1. Go to Events > Settings and set the date format as "j \d\e F \d\e Y"
2. Create an event
3. Make sure the shown date in the editor looks like "DAY de MONTH de YEAR"

### Changelog Entry

> Fixed - Bug fix Event date display in the editor when using escaped characters in date or time format

### Credits

Props @carstingaxion 

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/GatherPress/gatherpress/blob/main/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

**Comment by carstingaxion** [comment]
It's a little bit annoying to wait for php unit tests to finish, when the only changed files were JS files.
We should consider some separation at this point. 

**For the environment** :palm_tree: :deciduous_tree: :evergreen_tree: !

File: src/helpers/datetime.js [hunk 1]
```diff
@@ -461,10 +461,16 @@ export function convertPHPToMomentFormat(format) {
...
 		.split('')
-		.map((chr) => (chr in replacements ? replacements[chr] : chr))
+		.map((chr, index, elements) => {
+			// Allow the format string to contain escaped chars, like ES or DE needs
+			const last = elements[index - 1];
+			if (chr in replacements && last !== '\\') {
+				return replacements[chr];
+			}
+			return chr;
+		})
```
Comment (mauteri) [inline src/helpers/datetime.js:473]: @carstingaxion can we add a unit test for this? You can find them [here](https://github.com/GatherPress/gatherpress/blob/main/test/unit/js/src/helpers/datetime.test.js#L276). Thanks!!

**Review by mauteri — CHANGES_REQUESTED** [review]

File: src/helpers/datetime.js (hunk 1 above)
Comment (carstingaxion) [reply src/helpers/datetime.js:473]: Please have a look, at my added test.

**Comment by carstingaxion** [comment]
Ok, all tests passed, but [the one DB issue](https://github.com/WordPress/plugin-check-action/issues/61) [comes up again](https://github.com/GatherPress/gatherpress/actions/runs/8942833183/job/24566239852?pr=637#step:8:220)

**Comment by carstingaxion** [comment]
>Ok, all tests passed, but https://github.com/WordPress/plugin-check-action/issues/61 [comes up again](https://github.com/GatherPress/gatherpress/actions/runs/8942833183/job/24566239852?pr=637#step:8:220)

... which is a little weird because this branch hasn't even the workflow-file available.

**Review by mauteri — APPROVED** [review]

**Pull request merged**